File: MyApi/Models/Recipe.cs
using System.ComponentModel;

namespace MyApi.Models;

public class Recipe
{
    public int Id { get; set; }
    public required string Name { get; set; }

    public required string Description { get; set; }

    public ICollection<Category> Categories { get; set; } = [];

    public int PreparationTime { get; set; } // in minutes

    public int ServesCount { get; set; }
    public required string Ingredients { get; set; }
    public required string Instructions { get; set; }
    public string? ImagePath { get; set; }
}

File: MyClient/src/types/recipe.ts
export interface Recipe {
  id: number;
  name: string;
  description: string;
  category: string;
  preparationTime: number;
  servesCount: number;
  ingredients: string;
  instructions: string;
  imagePath?: string;
}

export interface ScannedRecipeResult {
  name: string;
  description: string;
  category: string;
  ingredients: string;
  instructions: string;
  detectedLanguage: string;
}
